### PR TITLE
Fix #8221: Subtitle - No Bridge Type Given Size

### LINKED ISSUE #8221: No error message subtitle for if no bridge types available for target length < max bridge length
## Version of OpenTTD
1.10.2

## Expected result
The error message should appear with any of the following subtitles:
* "...bridge too long" (same as if bridge was over the length limit specified by "Max bridge length")
* "...no bridge types available" (new message)

## Actual result
The error message ("can't build bridge here") does not render a subtitle.

## Steps to reproduce
1. Install and load the NewGRF "Expensive, Short, and Slow Bridges" (pb_bridges.grf).
2. Also install and load an early vehicle NewGRF (such as eGRVTS 2).
3. Set "Max bridge length" to 64 tiles or more.
4. Start a game in between 1890 and 1950.
5. Try to build a bridge greater than ~14 tiles long. It will fail but the error message will not have a subtitle.

Alternative steps to reproduce:
1. Download the scenario "Trans-Siberian Railway (1890)".
2. Select the scenario from within the "Play Scenario" menu and let it download any missing NewGRfs.
3. Start the scenario.
4. Try to build a bridge greater than ~14 tiles long. It will fail but the error message will not have a subtitle.

### PR DESCRIPTION
**Pull request by TrevorShelton**
The error box for going over max the lengths for each of the bridge types was given the "...bridge too long" subtitle.

### CODE REVIEW

File: src/bridge_gui.cpp [hunk 1]
```diff
@@ -426,9 +426,12 @@ void ShowBuildBridgeWindow(TileIndex start, TileIndex end, TransportType transpo
 			default: break;
 		}
 
+		bool any_available = 0;
```
Comment (glx22) [inline src/bridge_gui.cpp]:
```suggestion
		bool any_available = false;
```

File: src/bridge_gui.cpp [hunk 2]
```diff
@@ -437,8 +440,14 @@ void ShowBuildBridgeWindow(TileIndex start, TileIndex end, TransportType transpo
 				/* Add to terraforming & bulldozing costs the cost of the
 				 * bridge itself (not computed with DC_QUERY_COST) */
 				item.cost = ret.GetCost() + (((int64)tot_bridgedata_len * _price[PR_BUILD_BRIDGE] * item.spec->price) >> 8) + infra_cost;
+				any_available = 1;
```
Comment (glx22) [inline src/bridge_gui.cpp]:
```suggestion
				any_available = true;
```

**Review by glx22 — COMMENTED** [review]
Just a quick read, didn't check the actual change.

File: src/bridge_gui.cpp [hunk 3]
```diff
@@ -426,9 +426,12 @@ void ShowBuildBridgeWindow(TileIndex start, TileIndex end, TransportType transpo
 			default: break;
 		}
 
+		bool any_available = 0;
+		CommandCost type_check;
 		/* loop for all bridgetypes */
 		for (BridgeType brd_type = 0; brd_type != MAX_BRIDGES; brd_type++) {
-			if (CheckBridgeAvailability(brd_type, bridge_len).Succeeded()) {
+			type_check = CheckBridgeAvailability(brd_type, bridge_len);
+			if (type_check.Succeeded()) {
```
Comment (LordAro) [inline src/bridge_gui.cpp:434]: I don't see a reason for this change - type_check isn't used anywhere new

**Review by LordAro — APPROVED** [review]

**Pull request merged**